src/core/symbolFactories.ts

```diff
@@ -4,41 +4,56 @@
  */
 
 import * as d3 from "d3";
+import * as d3Shape from "d3-shape";
 
 /**
  * A SymbolFactory is a function that takes in a symbolSize which is the edge length of the render area
- * and returns a string representing the 'd' attribute of the resultant 'path' element
+ * and returns a d3 symbol generator.
  */
-export type SymbolFactory = (symbolSize: number) => string;
+export type SymbolFactory = (symbolSize: number) => d3Shape.Symbol<any, any>;
 
 export function circle(): SymbolFactory {
-  return (symbolSize: number) => d3.symbol().type(d3.symbolCircle).size(Math.PI * Math.pow(symbolSize / 2, 2))(null);
+  return (symbolSize: number) => {
```

Comment:
We're using d3 incorrectly here. We should create one d3.symbol() *per series*, not one per datapoint. We're creating 100k d3.symbol() objects and using it once per iteration each.